CMakeLists.txt

```diff
@@ -15,4 +15,4 @@ add_subdirectory_ifdef(CONFIG_MPSL                    mpsl)
 add_subdirectory_ifdef(CONFIG_BSD_LIBRARY             bsdlib)
 add_subdirectory_ifdef(CONFIG_NRFXLIB_CRYPTO          crypto)
 add_subdirectory_ifdef(CONFIG_NORDIC_SECURITY_BACKEND nrf_security)
-add_subdirectory_ifdef(CONFIG_ZIGBEE                  zboss)
+add_subdirectory_ifdef(CONFIG_APP_LINK_WITH_ZBOSS     zboss)
```

Comment:
Are you sure you want this dependency ?
What if downstream user creates an intermediate Zephyr library, and then links his app to that lib.
Then he won't need to link the app itself with Zboss, and can just link to his intermediate lib.